systemd: set withTpm2Tss and withUserDb to true on musl

yu-re-ka opened the pull request:
When we initially applied the openembedded patchset to make systemd build with musl, these options had to be disabled for it to work. Now they seem to work fine, so re-enabling.

Sending to staging because iproute2 is a dependency of tpm2tss and this is only fixed in staging by #190957 

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

alyssais commented:
@OfBorg build pkgsMusl.systemd

flokli commented:
This will conflict with https://github.com/NixOS/nixpkgs/pull/191032, or should be revisited after - should `withUtmp` be set to false if building for musl?

alyssais commented:
> should `withUtmp` be set to false if building for musl?

It should be, because musl doesn't implement utmp.  It looks like that's what https://github.com/NixOS/nixpkgs/pull/191032 does?

yu-re-ka commented:
Yes, utmp should stay disabled on musl. Only withTpm2Tss and withUserDb can be enabled.

We can just leave this open until #191032 reaches staging

flokli commented:
I tried cherry-picking this in on top of current master, but `iproute2` in the closure fails to build: 

```
   48 | #define IFF_VOLATILE (IFF_LOOPBACK|IFF_POINTOPOINT|IFF_BROADCAST| \
      | 
    CC       ipmptcp.o
    CC       iplink_bareudp.o
    CC       iplink_wwan.o
    CC       ipioam6.o
    CC       iplink_amt.o
    CC       iplink_batadv.o
    CC       iplink_gtp.o
    CC       iplink_virt_wifi.o
    CC       ipstats.o
    CC       rtmon.o
ipstats.c: In function ‘ipstats_show_64’:
ipstats.c:143:48: warning: implicit declaration of function ‘MIN’ [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]
  143 |                 memcpy(__dest, RTA_DATA(__at), MIN(__at_sz, __var_sz)); \
      |                                                ^~~
ipstats.c:157:9: note: in expansion of macro ‘IPSTATS_RTA_PAYLOAD’
  157 |         IPSTATS_RTA_PAYLOAD(stats, at);
      |         ^~~~~~~~~~~~~~~~~~~
    LINK     rtmon
    LINK     ip
/nix/store/jvsyjspqyrj6h6585d66k6xg3dyap5zc-binutils-2.38/bin/ld: ipstats.o: in function `ipstats_show_64':
ipstats.c:(.text+0x245): undefined reference to `MIN'
/nix/store/jvsyjspqyrj6h6585d66k6xg3dyap5zc-binutils-2.38/bin/ld: ipstats.o: in function `ipstats_stat_desc_show_l3_stats':
ipstats.c:(.text+0xe02): undefined reference to `MIN'
/nix/store/jvsyjspqyrj6h6585d66k6xg3dyap5zc-binutils-2.38/bin/ld: ipstats.o: in function `ipstats_stat_desc_show_afstats_mpls':
ipstats.c:(.text+0x1bc0): undefined reference to `MIN'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:29: ip] Error 1
make: *** [Makefile:76: all] Error 2
error: builder for '/nix/store/m2ax47lgn56khml5bc2gaw4rgzp8mqrz-iproute2-5.19.0.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/8sa4nc2sa278nld6sdfchqphg4nqwglc-tpm2-tss-3.2.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qk61n3ddi5xd7lz0c6v9krzv49b5fbfl-systemd-251.4.drv' failed to build
```

Is this fixed on staging?

yu-re-ka commented:
> Is this fixed on staging?

Yes, see the initial comment. https://github.com/NixOS/nixpkgs/pull/190957

flokli commented:
Yes, sorry, I somehow missed that :flushed:

With that cherry-picked too, I successfully built `pkgsMusl.systemd`. LGTM, but please resolve the conflicts so this can be merged :-)

yuyuyureka committed:
systemd: set withTpm2Tss and withUserDb to true on musl

When we initially applied the openembedded patchset to make systemd
build with musl, these options had to be disabled for it to work.
Now they seem to work fine, so re-enabling.

yu-re-ka commented:
rebased on staging
